### ISSUE TITLE
API keys are not saved in settings

### ISSUE THREAD

**Issue by WEZANGO** [cogentapps/chat-with-gpt#103]
I am using docker container and saved all API keys in my account, but every time I start a new session and log in to my account, all API keys are gone and needs to be entered again each time.

**Comment by catgirl69** [comment]
creat additional file then

**Comment by Arche151** [comment]
I have the same issue!

**Comment by febriansasi** [comment]
yes, me too, it missing if i close the tab on the browser

**Comment by ErfanDL** [comment]
same issue here how can I save the API key ? after refresh the page I lost all API keys

**Comment by McFex** [comment]
#metoo
any news on that?

**Comment by cogentapps** [comment]
Please let me know if this happens on the new version.

You can also store your API keys on the server now: https://github.com/cogentapps/chat-with-gpt#store-your-api-keys-on-the-server

**Comment by McFex** [comment]
reporting back - no issues for me on the new version.
I can save all keys through whatever button I choose (meaning button in the middle or settings wheel at the top or sidebar-email-user settings button).

**Comment by ErfanDL** [comment]
thanks solved the problem with new version but image size is huge !

**Comment by febriansasi** [comment]
The old version image only, 653MB. The update 6.5GB. It's using CUDA. Can I ask why? Isn't it just frontend UI for chatgpt.

**Comment by cogentapps** [comment]
Yeah, I moved to a CUDA base because I'm experimenting with adding a self-hosted TTS engine on the server, but since that hasn't been published yet a non-CUDA base image would make more sense.

**Comment by tha23rd** [comment]
> Yeah, I moved to a CUDA base because I'm experimenting with adding a self-hosted TTS engine on the server, but since that hasn't been published yet a non-CUDA base image would make more sense.

Hi - just wondering if this "light" image has been worked on? Would also appreciate an image not based on CUDA
